Tests/SwiftDependencyGraphsTests/Insert/InsertVertex.swift

```diff
@@ -0,0 +1,167 @@
+    #expect(result == (true, newVertex))
+  }
+
+  @Test("does not change the incoming edges") func incomingEdges() {
+    #expect(graph.incomingEdges == TestGraph.path().incomingEdges)
+  }
+
+  @Test("does not change the outgoing edges") func outgoingEdges() {
+    #expect(graph.outgoingEdges == TestGraph.path().outgoingEdges)
+  }
+
+  @Test("changes the vertices") func vertices() {
+    #expect(
+      graph.vertices == [
+        vertex1.id: vertex1,
+        vertex2.id: vertex2,
+        vertex3.id: vertex3,
+        vertex4.id: vertex4,
+        vertex5.id: vertex5,
+        newVertex.id: newVertex,
+      ])
+  }
+}
+
+// swiftlint:disable:next line_length
+@Suite("Inserting a new vertex with a unique id and a duplicate label into the graph") struct InsertVertexDuplicateLabelSuccessfullyTests {
+
+  var graph = TestGraph.path()
+  let newVertex = Vertex(id: 6, label: "5")
+  let result: (Bool, Vertex)
+
+  init() {
+    result = graph.insert(newVertex: newVertex)
+  }
+
+  @Test("returns that the insertion was successful and the inserted vertex") func returnValue() {
+    #expect(result == (true, newVertex))
+  }
+
+  @Test("does not change the incoming edges") func incomingEdges() {
+    #expect(graph.incomingEdges == TestGraph.path().incomingEdges)
+  }
+
+  @Test("does not change the outgoing edges") func outgoingEdges() {
+    #expect(graph.outgoingEdges == TestGraph.path().outgoingEdges)
+  }
+
+  @Test("changes the vertices") func vertices() {
+    #expect(
+      graph.vertices == [
+        vertex1.id: vertex1,
+        vertex2.id: vertex2,
+        vertex3.id: vertex3,
+        vertex4.id: vertex4,
+        vertex5.id: vertex5,
+        newVertex.id: newVertex,
+      ])
+  }
+}
+
+@Suite("Inserting a new vertex twice into the graph") struct InsertVertexTwiceTests {
+
+  var graph = TestGraph.path()
+  let newVertex = Vertex(id: -1)
+  let result: (Bool, Vertex)
+
+  init() {
+    _ = graph.insert(newVertex: newVertex)
+    result = graph.insert(newVertex: newVertex)
+  }
+
+  @Test("returns that the (second) insertion failed and the existing vertex") func returnValue() {
+    #expect(result == (false, newVertex))
+  }
+
+  @Test("does not change the incoming edges") func incomingEdges() {
+    #expect(graph.incomingEdges == TestGraph.path().incomingEdges)
+  }
+
+  @Test("does not change the outgoing edges") func outgoingEdges() {
+    #expect(graph.outgoingEdges == TestGraph.path().outgoingEdges)
+  }
+
+  @Test("changes the vertices") func vertices() {
+    #expect(
+      graph.vertices == [
+        vertex1.id: vertex1,
+        vertex2.id: vertex2,
+        vertex3.id: vertex3,
+        vertex4.id: vertex4,
+        vertex5.id: vertex5,
+        newVertex.id: newVertex,
+      ])
+  }
+}
+
+// swiftlint:disable:next line_length
+@Suite("Does not insert a vertex with a duplicate id and a duplicate label into the graph and") struct InsertVertexDuplicateIdAndLabelFailedTests {
```

Comment:
Produces a sentence, but I'm surprised to read this form (the ones above were more like expected)

```suggestion
@Suite("Inserting a vertex with existing ID and label into the graph") struct InsertVertexDuplicateIdAndLabelFailedTests {
```

maybe?

Reply:
I like the new sentence more. It fits more the form of the other sentences in the file.